swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt

```diff
@@ -0,0 +1,198 @@
+      }.toSet()
+
+    orcaService.orchestrate(
+      OrchestrationRequest(
+        application = application,
+        job = listOf(
+          OrcaJob(
+            type = "deleteLaunchConfiguration",
+            context = mutableMapOf(
+              "credentials" to workConfiguration.account.name,
+              "launchConfigurationNames" to launchConfigurationNames,
+              "cloudProvider" to AWS,
+              "region" to workConfiguration.location
+            )
+          )
+        ),
+        description = "Deleting LaunchConfigurations: $launchConfigurationNames"
+      )
+    ).let { taskResponse ->
+      val completeEvent = TaskCompleteEventInfo(
+        action = Action.DELETE,
+        markedResources = markedResources,
+        workConfiguration = workConfiguration,
+        submittedTimeMillis = clock.instant().toEpochMilli()
+      )
+
+      taskTrackingRepository.add(taskResponse.taskId(), completeEvent)
+    }
+  }
+
+  override fun handles(workConfiguration: WorkConfiguration): Boolean {
+    return workConfiguration.resourceType == LAUNCH_CONFIGURATION && workConfiguration.cloudProvider == AWS &&
+      rulesEngine.getRules(workConfiguration).isNotEmpty()
+  }
+
+  override fun getCandidates(workConfiguration: WorkConfiguration): List<AmazonLaunchConfiguration>? {
+    val params = Parameters(
+      account = workConfiguration.account.accountId!!,
+      region = workConfiguration.location,
+      environment = workConfiguration.account.environment
+    )
+
+    return aws.getLaunchConfigurations(params)
+  }
+
+  override fun getCandidate(
+    resourceId: String,
+    resourceName: String,
+    workConfiguration: WorkConfiguration
+  ): AmazonLaunchConfiguration? {
+    val params = Parameters(
+      id = resourceId,
+      account = workConfiguration.account.accountId!!,
+      region = workConfiguration.location,
+      environment = workConfiguration.account.environment
+    )
+
+    return aws.getLaunchConfiguration(params)
+  }
+
+  override fun preProcessCandidates(
+    candidates: List<AmazonLaunchConfiguration>,
+    workConfiguration: WorkConfiguration
+  ): List<AmazonLaunchConfiguration> {
+    return candidates
+      .also {
+        checkReferences(
+          launchConfigurations = it,
+          params = Parameters(
+            account = workConfiguration.account.accountId!!,
+            region = workConfiguration.location,
+            environment = workConfiguration.account.environment
+          )
+        )
+      }
+  }
+
+  /**
+   * Checks references for:
+   * Server Groups
+   */
+  private fun checkReferences(launchConfigurations: List<AmazonLaunchConfiguration>?, params: Parameters) {
+    if (launchConfigurations.isNullOrEmpty()) {
+      return
+    }
+
+    log.debug("checking references for {} launch configs. Parameters: {}", launchConfigurations.size, params)
+    val elapsedTimeMillis = measureTimeMillis {
+      setServerGroupReferences(launchConfigurations, params)
```

Comment:
can this throw an exception ? do we need to wrap this with a try/catch ?

Reply:
Exceptions would bubble up and get handled in `AbstractResourceTypeHandler`